scripts/missions/amk/13_A_Challenge_You_Could_Be_a_Winner.lua

```diff
@@ -354,16 +359,91 @@ mission.sections =
 
         [xi.zone.CASTLE_ZVAHL_BAILEYS] =
         {
+            ['Shadowy_Pillar'] =
+            {
+                onTrigger = function(player, npc)
+                    if not player:needToZone() then
+                        return mission:progressEvent(100, 1)
+                    else
+                        return mission:progressEvent(100, 2)
+                    end
+                end,
+            },
+
+            ['Flame_of_Fate'] =
+            {
+                onTrigger = function(player, npc)
+                    player:delStatusEffect(xi.effect.LEVEL_RESTRICTION)
+
+                    if not player:needToZone() then
+                        return mission:progressEvent(101, 4)
+                    elseif player:getLocalVar('Mission[10][12][p3]timeLimit') < os.time() then
+                        return mission:progressEvent(101, 1)
+                    else
+                        return mission:progressEvent(101, 2)
+                    end
+                end,
+            },
 
+            onEventFinish =
+            {
+                [100] = function(player, csid, option, npc)
+                    if option == 1 then
+                        -- Start run
+                        player:needToZone(true)
+                        player:setLocalVar('Mission[10][12][p3]timeLimit', os.time() + utils.minutes(8))
```

Comment:
This can be replaced with mission:setLocalVar

Reply:
I have since learned more comprehensively about how those work, will update.

Reply:
All set